rand/rand.go

```diff
@@ -0,0 +1,107 @@
+package rand
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math"
+	"math/big"
+)
+
+const (
+	// DefaultCharset defines the default characters
+	DefaultCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+	// DefaultLength defines the default length for random string
+	DefaultLength = 10
+)
+
+func Number() (int64, error) {
+	n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		return 0, fmt.Errorf("failed to generate random number: %w", err)
+	}
+	return n.Int64(), nil
+}
+
+// NumberInRange generates a random number between min and max
+func NumberInRange(min, max int64) (int64, error) {
```

Comment:
good to add an early case: If min == max

Reply:
Yeah that makes sense, thanks!